pkg/machines/components/create-vm-dialog/Password.jsx

```diff
@@ -0,0 +1,125 @@
+        super(props);
+        this.state = {
+            passwordQuality: "",
+            warningQuality: "",
+            displayWarningQual: false
+        };
+        this.handleChangePassword = this.handleChangePassword.bind(this);
+        this.handleWarnings = this.handleWarnings.bind(this);
+        this.validateQuality = this.validateQuality.bind(this);
+        this.passwordQualityFail = this.passwordQualityFail.bind(this);
+        this.passwordQualityPass = this.passwordQualityPass.bind(this);
+    }
+
+    handleChangePassword(event) {
+        this.setState({
+            displayWarningQual: false
+        });
+        this.props.onValueChanged(event.target.value);
+
+        if (event.target.value !== "") {
+            this.validateQuality(event.target.value);
+        } else {
+            this.setState({
+                passwordQuality: "",
+                warningQuality: ""
+            });
+        }
+    }
+
+    handleWarnings() {
+        setTimeout(() => {
+            this.setState({ displayWarningQual: true });
+        }, 300);
+    }
+
+    validateQuality(password) {
+        const dfd = cockpit.defer();
+        cockpit
+                .spawn("/usr/bin/pwscore", { err: "message" })
+                .input(password)
+                .done(content => {
+                    const quality = parseInt(content, 10);
+                    if (quality === 0) {
+                        this.passwordQualityFail(messages.weakPassword);
+                        dfd.reject(new Error(messages.weakPassword));
+                    } else if (quality <= 33) {
+                        this.passwordQualityPass("weak", password);
+                        dfd.resolve("weak");
+                    } else if (quality <= 66) {
+                        this.passwordQualityPass("okay", password);
+                        dfd.resolve("okay");
+                    } else if (quality <= 99) {
+                        this.passwordQualityPass("good", password);
+                        dfd.resolve("good");
+                    } else {
+                        this.passwordQualityPass("excellent", password);
+                        dfd.resolve("excellent");
+                    }
+                })
+                .fail(ex => {
+                    this.passwordQualityFail(ex.message || messages.rejectedPassword);
+                });
+        return dfd.promise();
+    }
+
+    passwordQualityFail(message) {
+        this.setState({ passwordQuality: "weak", warningQuality: message });
+        this.props.onValueChanged(undefined);
+    }
+
+    passwordQualityPass(quality, password) {
+        this.setState({ passwordQuality: quality, warningQuality: "" });
+        this.props.onValueChanged(password);
+    }
+
+    render() {
+        const passwordInvalid = this.state.warningQuality && this.state.displayWarningQual;
+
+        return (
+            <>
+                <FormGroup validationState={passwordInvalid ? "error" : undefined}>
```

Comment:
Does a false validationState cause the dialog to reject the password? That's not quite clear to me from looking at the code. IMHO it shouldn't -- warning is okay, but if you want to create a test VM with password "foobar" (who would do that!? :grin: ), we shouldn't second-guess the user.

Reply:
It only shows a warning now, that's good. Thanks!